irods/manager/data_object_manager.py

```diff
@@ -628,9 +628,13 @@ def truncate(self, path, size, **options):
             response = conn.recv()
 
 
-    def replicate(self, path, resource=None, **options):
+    def replicate(self, path, resource = None, **options):
         if resource:
             options[kw.DEST_RESC_NAME_KW] = resource
+        else:
+            def_resc = getattr(self.sess,'default_resource',None)
+            if def_resc:
+                options[kw.DEF_RESC_NAME_KW] = def_resc
```

Comment:
Should this be `DEST_RESC_NAME_KW` instead of `DEF_RESC_NAME_KW`?

I understand why `DEF_RESC_NAME_KW` is used, but I don't see a reason to continue supporting it.

Do the keywords result in different behavior?

Reply:
We should probably have that discussion in #459 if we aren't planning to support the keyword in PRC. We should probably have a discussion in an irods/irods issue if we are planning to drop support for the keyword in the replication API.

Reply:
Oops, didn't answer the question, sorry. The keywords do result in different behavior for replication.

Reply:
Got it. I just wanted to know if the keywords result in different behavior.

Totally agree with everything you said.

Definitely not a blocker for this PR.